src/modules/CRUD/components/DataTable/Footer/index.js

```diff
@@ -1,9 +1,9 @@
 import { withNamespaces } from 'react-i18next';
-import { withPageSize } from '../../../services/UserSettingsProvider';
+import { connectUserSettingsProvider } from '../../../services/UserSettingsProvider';
 import compose from '../../../../../utils/compose';
 import Footer from './Footer';
 
 export default compose(
-  withPageSize(),
+  connectUserSettingsProvider('tableFilters', 'setTableFilters', 'pageSize', 'setPageSize'),
```

Comment:
Why not using a `withTableFilter()` _hoc_ like `withPageSize()` ?

Reply:
It may be better for readability purposes, but I think that calling the same HOC twice is a bit of overkill, isn't it?

Reply:
I didn't do any test about that point, but yes, it's a point. I don't have any idea if setting multiple times the same HOC may be an issue or not. It should be checked...

But I think that the real pattern that we should use in this case is custom hooks *(`const [pageSize, setPageSize] = usePageSize();`)* instead of HOCs. But we do not have define any guideline about that for now.

So, I would suggest you to quickly test *if* setting the same HOCs seems implying perf/rendering issues.
If not, `withThisData()` is clearly best for DX and maintainability. If any suspicion of negative impact, keep your initial choice.